<a href="https://colab.research.google.com/github/Akshay0796/Python-for-DS/blob/main/Buyer_Ratio_Assignment_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

We Have to check that the Male & Female ratio across the regions are equal or not.
For this we will use Chi2 test.

Ho = All proportions are equal.

Ha = Not all proportions are equal.

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import chi2_contingency

In [3]:
df = pd.read_csv('/content/BuyerRatio.csv')

In [4]:
df

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [9]:
# we have to make dimensional arrays
sales = np.array([[50,142,131,70],[435,1523,1356,750]])
sales

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [10]:
#Chi2 test
chi2_contingency(sales)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

From above chi2 test result P value is 0.6603094907091882.

Chi2 statistical value is 1.595945538661058. Degree of freedom is 3.

And we got expected frequency of values.

From above result we can see that the P= 0.6603094907091882 > 0.05. Thus we accept the null hypothesis.
we can say that the Male and Female buyer ratios are similar across regions and are not related.